lib/config.go

```diff
@@ -360,8 +359,19 @@ func (c *Config) ToFile(path string) error {
 
 // DefaultConfig returns the default configuration for crio.
 func DefaultConfig() (*Config, error) {
-	registries, _ := sysregistries.GetRegistries(&types.SystemContext{})
-	insecureRegistries, _ := sysregistries.GetInsecureRegistries(&types.SystemContext{})
+	registries, err := sysregistriesv2.UnqualifiedSearchRegistries(&types.SystemContext{})
```

Comment:
@mtrmac, can we log the errors? Ultimately, I would prefer to return the error but we could defer that to another PR. Currently, there's no (easy) way a user/admin could figure out if the config is loaded; only via the http API or when pulling.

Reply:
From the PR description:
> NOTE: It's unclear to me why `lib.DefaultConfig` reads `registries.conf` for defaults; AFAICS the values end up only passed in `pkg/storage.GetImage`<s>`Sources`</s>`Service`, which already defaults the configuration to the one in `sysregistriesv2`, so explicitly creating that default in the callers as well seems rather redundant.

So, whether or not this fails, the values read at this point are not actually indicative of what CRI-O will later use. I suppose `GetImageService` could log its final configuration, and that would be representative for the search path (which should ~never be used because it doesn’t work with credentials), but not quite for the insecure registry list (which is also implicitly applied deep inside c/image/docker).

Reply:
If you don’t mind, I’ll add logging here in a separate PR, so that this PR is consistent between the 1.14 and master branches, and there are no surprises when updating from 1.14.

Reply:
… hum, actually, there already is a (racy, partial) check in `RuntimeConfig.Validate`.

Looking further, it seems that the `DefaultConfig` value is supposed to be useful for `cri-o config --default`, to write a configuration file equivalent to the built-in defaults. The way I think about it, that really should not contain any copies of data from `registries.conf`; we should default to just using the `registries.conf` values unless the user _explicitly_ decides to override them.

If you agree, I’ll rip these calls out of `DefaultConfig` entirely, and we’ll rely on `GetImageService` calling `sysregistriesv2.*` if the config has not been explicitly set up by the user; the call in  `RuntimeConfig.Validate` can continue to serve as a sanity check that `registries.conf` is valid, at least at the time CRI-O is starting.

Reply:
I am cool with merging this as is. As you suggested, we can figure out the details of that in another PR.